pyuvsim/simsetup.py

```diff
@@ -0,0 +1,388 @@
+from pyuvdata import UVBeam, UVData
+import numpy as np
+import yaml
+import os
+import pyuvdata.utils as uvutils
+
+# Utilities for setting up simulations for parameter files,
+# and for generating parameter files from uvfits files
+
+
+def check_file_exists_and_increment(filepath):
+    """
+        Given filepath (filename + path), check if it exists. If so, add a _1
+        at the end. etc.
+    """
+    if os.path.exists(filepath):
+        filepath += "_0"
+    else:
+        return filepath
+    n = 1
+    while os.path.exists(filepath):
+        filepath = filepath[:-2] + "_" + str(n)
+        n += 1
+    return filepath
+
+
+def parse_layout_csv(layout_csv):
+    """ Interpret the layout csv file """
+
+    header = open(layout_csv, 'r').readline()
+    header = [h.strip() for h in header.split()]
+    dt = np.format_parser(['a10', 'i4', 'i4', 'f8', 'f8', 'f8'],
+                          ['name', 'number', 'beamid', 'e', 'n', 'u'], header)
+
+    return np.genfromtxt(layout_csv, autostrip=True, skip_header=1,
+                         dtype=dt.dtype)
+
+
+def initialize_uvdata_from_params(param_dict):
+    """
+        Construct a uvdata object from parameters in a valid yaml file.
+
+        Arguments:
+            param_dict : Dictionary of parameters read in.
+                    Any uvdata parameters may be passed in through here.
+        Returns:
+            uv_obj, beam_list, beam_ids
+    """
+
+    # Parse telescope parameters
+    tele_params = param_dict['telescope']
+
+    telescope_config_path = tele_params['telescope_config_path']
+    layout_csv = tele_params['array_layout']
+
+    if not os.path.exists(telescope_config_path):
+        path = os.path.dirname(param_dict['config_path'])
+        telescope_config_path = os.path.join(path, telescope_config_path)
+
+    if not os.path.exists(layout_csv):
+        path = os.path.dirname(param_dict['config_path'])
+        layout_csv = os.path.join(path, layout_csv)
+
+    ant_layout = parse_layout_csv(layout_csv)
+
+    with open(telescope_config_path, 'r') as yf:
+            telparam = yaml.safe_load(yf)
+            tloc = telparam['telescope_location'][1:-1]  # drop parens
+            tloc = map(float, tloc.split(","))
+            tloc[0] *= np.pi/180.
+            tloc[1] *= np.pi/180.   # Convert to radians
+            telparam['telescope_location'] = uvutils.XYZ_from_LatLonAlt(*tloc)
+
+    param_dict.update(telparam)
+    E, N, U = ant_layout['e'], ant_layout['n'], ant_layout['u']
+    antnames = ant_layout['name']
+    beam_ids = ant_layout['beamid']
+    beam_list = []
+    for beamID in np.unique(beam_ids):
+        path = telparam['beam_paths'][beamID]
+        uvb = UVBeam()
+        uvb.read_beamfits(path)
```

Comment:
I think this means that you cannot pass Nic's CST beam files to this code. Is that what we want?